src/components/menu/menu.svelte

```diff
@@ -254,23 +256,29 @@
     display: revert;
   }
 
-  :global .leo-menu-popup ::slotted(leo-menu-item:hover),
-  :global .leo-menu-popup ::slotted(leo-option:hover),
-  :global .leo-menu-popup > leo-menu-item:hover,
-  :global .leo-menu-popup > leo-option:hover {
+  :global :where(.leo-menu-popup) ::slotted(leo-menu-item:hover),
+  :global :where(.leo-menu-popup) ::slotted(leo-option:hover),
+  :global :where(.leo-menu-popup) > leo-menu-item:hover,
+  :global :where(.leo-menu-popup) > leo-option:hover {
     background: var(--leo-color-container-highlight);
   }
 
-  :global .leo-menu-popup ::slotted(*[aria-selected]),
-  :global .leo-menu-popup ::slotted(*:active),
-  :global .leo-menu-popup > *[aria-selected],
-  :global .leo-menu-popup > *:active {
+  :global :where(.leo-menu-popup) ::slotted(leo-option[aria-selected]),
+  :global :where(.leo-menu-popup) ::slotted(leo-menu-item[aria-selected]),
+  :global :where(.leo-menu-popup) ::slotted(leo-option:active),
+  :global :where(.leo-menu-popup) ::slotted(leo-menu-item:active),
+  :global :where(.leo-menu-popup) > leo-option[aria-selected],
+  :global :where(.leo-menu-popup) > leo-menu-item[aria-selected],
+  :global :where(.leo-menu-popup) > leo-option:active,
+  :global :where(.leo-menu-popup) > leo-menu-item:active {
     background: var(--leo-color-container-interactive);
     color: var(--leo-color-text-interactive);
   }
 
-  :global .leo-menu-popup ::slotted(*:focus-visible),
-  :global .leo-menu-popup > *:focus-visible {
+  :global :where(.leo-menu-popup) ::slotted(leo-option:focus-visible),
+  :global :where(.leo-menu-popup) ::slotted(leo-menu-item:focus-visible),
+  :global :where(.leo-menu-popup) > leo-option:focus-visible,
+  :global :where(.leo-menu-popup) > leo-menu-item:focus-visible {
```

Comment:
Though more verbose, this makes it easier to have other content in the menu without having to reset their color/background color.